### DISCUSSION TITLE
NextJS API routes are missing from Vercel's build outputs when folder contains index.js/ts

### DISCUSSION THREAD

**Discussion by angusryer** [vercel/community#185] [Help] [answered]
I'm using NextJS /api routes and deploying to Vercel.

It seems that wherever I have an `index.ts` file located within the `/api` folder structure, Vercel will replace the contents of its containing folder (including `/api` itself) with a single file. All of the sibling route files will be ignored.

Oddly enough, the folder that contains the index file will remain--but without contents.

Ex:

```
/api/index.ts
/api/mock1/index.ts
/api/mock1/mock1-endpoint.ts
/api/mock2/index.ts
/api/mock2/mock2-endpoint.ts
```

This will produce an output of:

```
/api // empty folder
λ api
```
----
- Note: you'll have to deploy this code to see the output in Vercel's dashboard
----
Reproducible code: https://github.com/angusryer/missing-lambdas-vercel-nextjs

Deployed example: https://missing-lambdas-vercel-nextjs.vercel.app/

##### The api routes _are_ visible as seen in the 'source' tab of the production deployment:

[![source files available][1]][1]

##### But they are not available in the 'output' tab:

[![no output files][2]][2]

Does anyone know why this happens and ways to get around it?

  [1]: https://i.stack.imgur.com/aXLYb.png
  [2]: https://i.stack.imgur.com/N8HPe.png

**Comment by Pieparker** [comment, marked as answer]
This conversation appears to have gone quiet. To keep discussions on topic and to prevent the resurrection of old topics, this discussion has been closed. If issues persist or a future reader stumbles upon this discussion and is exhibiting the same problem, please create a new discussion.
